Homework/Program.cs

```diff
@@ -1,42 +1,102 @@
+        public void DisplayCarInfo()
         {
-            // Načítání údajů pro osobu
-            Console.WriteLine($"Zadejte údaje pro osobu {i}:");
+            Console.WriteLine($"Brand: {Brand}, Model: {Model}, Year: {Year}, Mileage: {Mileage} km");
+        }
+    }
+
+    // Třída reprezentující knihu
+    public class Book
+    {
+        // Vlastnosti (properties)
+        public string Title { get; set; }
+        public string Author { get; set; }
+        public int Pages { get; set; }
+        public int CurrentPage { get; set; }
 
-            // Získání údajů od uživatele
-            Console.Write("Jméno: ");
-            string firstName = Console.ReadLine();
+        // Konstruktor, který inicializuje všechny vlastnosti
+        public Book(string title, string author, int pages)
+        {
+            Title = title;
+            Author = author;
+            Pages = pages;
+            CurrentPage = 0; // Výchozí hodnota pro aktuální stránku
+        }
 
-            Console.Write("Příjmení: ");
-            string lastName = Console.ReadLine();
+        // Metoda pro přidání přečtených stran
+        public void Read(int pages)
+        {
+            CurrentPage += pages;
+            if (CurrentPage > Pages)
+            {
+                CurrentPage = Pages; // Pokud přečteme více než je počet stran, nastavíme na maximum
+            }
+        }
 
-            Console.Write("Věk: ");
-            int age = int.Parse(Console.ReadLine());
+        // Metoda pro zobrazení pokroku v knize
+        public void DisplayProgress()
+        {
+            Console.WriteLine($"Reading '{Title}' by {Author}: {CurrentPage}/{Pages} pages read");
+        }
+    }
 
-            Console.Write("Váha (kg): ");
-            double weight = double.Parse(Console.ReadLine());
+    class Program
+    {
+        static void Main(string[] args)
+        {
+            // Vytvoření objektu třídy Car
+            Car car1 = new Car("Porsche", "GT3-911", 2009, 25340);
+            Car car2 = new Car("Ferrari", "Monza SP1", 2018, 10356);
 
-            Console.Write("Výška (cm): ");
-            double height = double.Parse(Console.ReadLine());
+            // Použití metod třídy Car
+            car1.DisplayCarInfo();
+            car2.DisplayCarInfo();
+            car1.Drive(100);
+            car2.Drive(200);
+            car1.DisplayCarInfo();
+            car2.DisplayCarInfo();
 
-            // Zobrazit uživatelské informace
-            Console.WriteLine($"{firstName} {lastName} je starý(á) {age} let. Váží {weight} kilogramů a je {height} centimetrů vysoký/á.");
+            Console.WriteLine();
 
-            // Výpočet BMI
-            double heightInMeters = height / 100; // Převod výšky na metry
-            double bmi = weight / (heightInMeters * heightInMeters); // Vzorec pro BMI
+            // Vytvoření objektu třídy Book
+            Book book1 = new Book("Harry Potter And The Phillosophers Stone", "J.K. Rowling", 320;
```

Comment:
Tady chybí závorka při inicializaci objektu. Kód kvůli tomu nejde zkompilovat a spustit
```suggestion
            Book book1 = new Book("Harry Potter And The Phillosophers Stone", "J.K. Rowling", 320);
```